server/routes/storage.js

```diff
@@ -31,7 +31,7 @@ router.get('/decks', (req, res) => {
 
 router.post('/decks', (req, res) => {
   const { title, source } = req.body || {};
-  if (!title || typeof title !== 'string') {
+  if (!title || title.trim() === '' || typeof title !== 'string') {
```

Comment:
Good addition using `trim`, could simplify a little - `typeof title !== 'string'` covers null/undefined and title.trim() === ''` covers empty string, so don't need `!title` - could do:

```
if (typeof title !== 'string' || title.trim() === '')
```